### PR TITLE
Added types for existing snowball-stemmers package

### PR DESCRIPTION
**Pull request by ryanvolum**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/snowball-stemmers/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Type definitions for snowball-stemmers 0.6
+// Project: https://github.com/mazko/jssnowball#readme
+// Definitions by: Ryan Volum <https://github.com/me>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export class Stemmer {
+    stem(term: string): string;
+}
+
+export function newStemmer(locale: string): Stemmer;
```
Comment (sheetalkamat) [inline types/snowball-stemmers/index.d.ts]: Missing function algorithms as per https://www.npmjs.com/package/snowball-stemmers#example

File: types/snowball-stemmers/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Type definitions for snowball-stemmers 0.6
+// Project: https://github.com/mazko/jssnowball#readme
+// Definitions by: Ryan Volum <https://github.com/me>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export class Stemmer {
```
Comment (sheetalkamat) [inline types/snowball-stemmers/index.d.ts]: This cannot be class and needs to be interface. When I ran below code:
`
var snowballFactory = require('snowball-stemmers');
console.log(typeof snowballFactory.Stemmer);
`

It prints `undefined`. Noticing that there is no object named Stemmer on the factory.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Comment by ryanvolum** [comment]
Hey there @sheetalkamat! I made the changes you asked for and everything is passing. Please review and let me know if there is anything else I should change!

File: types/snowball-stemmers/index.d.ts (hunk 1 above)
Comment (ryanvolum) [reply types/snowball-stemmers/index.d.ts]: @sheetalkamat I've now included a function for algorithms

File: types/snowball-stemmers/index.d.ts (hunk 2 above)
Comment (ryanvolum) [reply types/snowball-stemmers/index.d.ts]: @sheetalkamat I've updated the class to be an interface

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**